app/js/checkAnswers.js

```diff
@@ -129,3 +129,83 @@ function displayResult(earnedPoints, earnedPercentage, answeredQuestions) {
     document.querySelector(".answered_questions").innerHTML = answeredQuestions
     document.querySelector(".score_percentage").innerHTML = earnedPercentage
 }
+
+/**
+ * function adds event listeners to .question and listens for click event within here
+ * it then updates the class of the span containing the question number allowing styling to be applied
+ *
+ */
+function addAnswerEventListeners() {
+    document.querySelectorAll('.question').forEach(function (input) {
+        input.addEventListener('click', function(e) {
+            if (e.target.tagName == 'INPUT') {
+                let nav = document.querySelector('#question-nav')
+                let id = parseInt(this.dataset['id']) - 1
+                nav.children[id].classList.add('answered-nav-box')
+            }
+        })
+    })
+}
+
+/**
+ * function removes current status from all questions and then adds current status
+ * to the current question allowing styling to be applied
+ *
+ * @param id is the id of the active question
+ *
+ */
+function trackActiveQuestion(id) {
+    let nav = document.querySelector('#question-nav')
+    for (let i = 0; i < nav.children.length; i++) {
```

Comment:
this is a really slow way of doing this. why not just do:
`document.querySelector('#question-nav .current-nav-box').classList.remove('current-nav-box')`

Reply:
refactored in de918e0